src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts

```diff
@@ -0,0 +1,160 @@
+            this.submission = this.participation.submissions[0];
+        }
+        // for normal exercises we track the number of allowed complaints
+        if (this.course?.complaintsEnabled) {
+            this.complaintService.getNumberOfAllowedComplaintsInCourse(this.course!.id!, this.exercise.teamMode).subscribe((allowedComplaints: number) => {
+                this.numberOfAllowedComplaints = allowedComplaints;
+            });
+        }
+        this.loadPotentialComplaint();
+        this.accountService.identity().then((user) => {
+            if (this.participation && this.participation.student && user?.id) {
+                this.isCurrentUserSubmissionAuthor = this.participation.student.id === user.id;
+            }
+        });
+
+        this.timeOfFeedbackRequestValid = this.isTimeOfFeedbackRequestValid();
+        this.timeOfComplaintValid = this.isTimeOfComplaintValid();
+        this.showComplaintsSection = this.isShowComplaintsSection();
+    }
+
+    /**
+     * Sets the complaint if a complaint and a valid result exists
+     */
+    loadPotentialComplaint(): void {
+        if (this.result && this.result.completionDate) {
+            this.complaintService
+                .findBySubmissionId(this.submission.id!)
+                .pipe(filter((res) => !!res.body))
+                .subscribe((res) => {
+                    this.complaint = res.body!;
+                });
+        }
+    }
+
+    /**
+     * This function is used to check whether the student is allowed to submit a complaint or not.
+     * For exams, submitting a complaint is allowed within the Student Review Period, see {@link isWithinExamReviewPeriod}.
+     *
+     * For normal course exercises, submitting a complaint is allowed within one week after the student received the
+     * result. If the result was submitted after the assessment due date or the assessment due date is not set, the completion date of the result is checked. If the result was
+     * submitted before the assessment due date, the assessment due date is checked, as the student can only see the result after the assessment due date.
+     */
+    private isTimeOfComplaintValid(): boolean {
+        if (this.result?.completionDate) {
+            if (this.isExamMode) {
+                return this.isWithinExamReviewPeriod();
+            }
+            return this.isCompletionDateWithinCourseReviewPeriod(this.result.completionDate);
+        }
+        return false;
+    }
+
+    /**
+     * Analogous to isTimeOfComplaintValid but exams cannot have more feedback requests.
+     */
+    private isTimeOfFeedbackRequestValid(): boolean {
+        if (!this.isExamMode && this.result?.completionDate) {
+            return this.isCompletionDateWithinCourseReviewPeriod(this.result.completionDate);
+        }
+        return false;
+    }
+
+    /**
+     * Determines whether or not to show the complaint section
+     */
+    private isShowComplaintsSection(): boolean {
```

Comment:
this sounds a bit odd... you we change it to `isComplaintSectionShown` or `showComplaintsSection`?

Reply:
I see your point. `isCOmplaintSectionShown` is wrong because I don't want to determine whether something is shown but whether to show something or not.
`showComplaintsSection` is ambiguous. It can have the meaning you want to have, or it can be an action.

That's why I added the `is` to stress that it is an evaluation function, not an action function.

Maybe something like `shouldShowComplaintsSection`? Sounds odd to me too, though.

Reply:
I think `shouldShowComplaintsSection` is fine...

Reply:
Changed it to `getComplaintsSectionVisibility()`. Does this work for you as well?